test/TestHelper.cpp

```diff
@@ -94,22 +95,43 @@ struct MissingFields : virtual Exception {};
 
 bigint const c_max256plus1 = bigint(1) << 256;
 
-ImportTest::ImportTest(json_spirit::mObject& _o, bool isFiller):
-	m_statePre(OverlayDB(), eth::BaseState::Empty, Address(_o["env"].get_obj()["currentCoinbase"].get_str())),
-	m_statePost(OverlayDB(), eth::BaseState::Empty, Address(_o["env"].get_obj()["currentCoinbase"].get_str())),
-	m_TestObject(_o)
+ImportTest::ImportTest(json_spirit::mObject& _o, bool isFiller, testType testTemplate):
+	m_statePre(OverlayDB(), eth::BaseState::Empty),
+	m_statePost(OverlayDB(), eth::BaseState::Empty),
+	m_testObject(_o)
 {
-	importEnv(_o["env"].get_obj());
-	importState(_o["pre"].get_obj(), m_statePre);
-	importTransaction(_o["transaction"].get_obj());
-
-	if (!isFiller)
+	if (testTemplate == testType::StateTests)
 	{
-		importState(_o["post"].get_obj(), m_statePost);
-		m_environment.sub.logs = importLog(_o["logs"].get_array());
+		importEnv(_o["env"].get_obj());
+		importTransaction(_o["transaction"].get_obj());
+		importState(_o["pre"].get_obj(), m_statePre);
+		if (!isFiller)
+		{
+			if (_o.count("post"))
+				importState(_o["post"].get_obj(), m_statePost);
+			else
+				importState(_o["postState"].get_obj(), m_statePost);
+			m_logsExpected = importLog(_o["logs"].get_array());
+		}
 	}
 }
 
+//executes an imported transacton on preState
+bytes ImportTest::executeTest()
```

Comment:
we will later (in another PR) refactor the test code, this will be moved, since it is not related to importing the test.